code/obj/item/device/flash_pistol.dm

```diff
@@ -0,0 +1,224 @@
+			disorient_time = src.flash_disorient_time)
+
+		M.update_burning(src.flash_burning)
+
+		// Log entry.
+		var/blind_msg_target = "!"
+		var/blind_msg_others = "!"
+		if (!blind_success)
+			blind_msg_target = " but your eyes are protected!"
+			blind_msg_others = " but [his_or_her(M)] eyes are protected!"
+		M.visible_message("<span class='alert'>[user] blinds [M] with \the [src][blind_msg_others]</span>", "<span class='alert'>[user] blinds you with \the [src][blind_msg_target]</span>")
+		logTheThing(LOG_COMBAT, user, "blinds [constructTarget(M,"combat")] with [src] at [log_loc(user)].")
+
+		// Some after attack stuff.
+		user.lastattacked = M
+		M.lastattacker = user
+		M.lastattackertime = world.time
+
+		return
+
+	update_icon()
+		if (src.loaded)
+			inventory_counter?.update_number(1)
+			src.icon_state = replacetext(src.icon_state, "-empty", "")
+		else
+			inventory_counter?.update_number(0)
+			if (!findtext(src.icon_state, "-empty"))
+				src.icon_state = "[src.icon_state]-empty"
+
+		if (src.hammer_cocked && !findtext(src.icon_state, "-c"))
+			src.icon_state = "[src.icon_state]-c"
+		else if (!src.hammer_cocked)
+			src.icon_state = replacetext(src.icon_state, "-c", "")
+
+
+/obj/item/flash_compound_bottle
+	name = "Flash Compound Bottle"
+	desc = "A small bottle containing a mixture specialised to create a vivid flash when used in a flash pistol."
+	icon = 'icons/obj/items/ammo.dmi'
+	icon_state = "flash_compound_bottle"
+	inventory_counter_enabled = TRUE
+	w_class = W_CLASS_SMALL
+	var/amount_left = 10
+	var/max_amount = 10
+
+	New()
+		. = ..()
+		inventory_counter?.update_number(amount_left)
+
+	attackby(obj/O, mob/user)
+		if(istype(O, /obj/item/flash_pistol))
+			O.Attackby(src, user)
+
+		else if(O.type == src.type)
+			var/obj/item/flash_compound_bottle/compound_bottle = O
+			if(compound_bottle.amount_left < 1)
+				user.show_text("There's no compound left in [compound_bottle.name].", "red")
+				return
+			if(src.amount_left >= src.max_amount)
+				user.show_text("[src] is full!", "red")
+				return
+
+			while ((compound_bottle.amount_left > 0) && (src.amount_left < src.max_amount))
+				compound_bottle.amount_left--
+				src.amount_left++
+
+			if ((compound_bottle.amount_left < 1) && (src.amount_left < src.max_amount))
+				compound_bottle.UpdateIcon()
+				src.UpdateIcon()
+				user.visible_message("<span class='alert'>[user] refills [src].</span>", "<span class='alert'>There wasn't enough compound left in [compound_bottle.name] to fully refill [src]. It only has [src.amount_left] uses remaining.</span>")
+				return
+
+			if ((compound_bottle.amount_left >= 0) && (src.amount_left == src.max_amount))
+				compound_bottle.UpdateIcon()
+				src.UpdateIcon()
+				user.visible_message("<span class='alert'>[user] refills [src].</span>", "<span class='alert'>You fully refill [src] with compound from [compound_bottle.name]. There are [compound_bottle.amount_left] uses left in [compound_bottle.name].</span>")
+				return
```

Comment:
Consistancy with other reloading messages. Flash compound is also not a reagent; I believe it would be misleading to describe it in terms often associated with transferring reagents.

Reply:
If other ammo transfer stuff works like this then that's fine 👍